tiatoolbox/dataloader/wsireader.py

```diff
@@ -1059,14 +1068,28 @@ class VirtualWSIReader(WSIReader):
 
     """
 
-    def __init__(self, input_img):
+    def __init__(
+        self,
+        input_img,
+        baseline_size=None,
+        mpp=None,
+        power=None,
+        level_downsamples=None,
+        level_dimensions=None,
+    ):
```

Comment:
why provide these manually ? we have the common meta class, should receive that one as input. The virtual will likely need to match the source WSI anyways, and the source has its meta readied.

Reply:
We may also need a func to allow fast swapping metadata.

Reply:
These were defined as arguments because it seemed a bit clunky to have to define a metadata object before calling the init but I can certainly see the sense.

RE the fast swapping. You can reassign the info with another, the `@info.setter` allows swapping out via simple assignment e.g. `wsi.info = WSIMeta(...)`.